### PR TITLE
Animation graph: experimentally support more types of variable

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

* This PR provides the supports of vector3/quaternion variables in animation graph. Variable of these new experimental types can be in future used in pose graph, as introduced in PR https://github.com/cocos/cocos-engine/pull/14779 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/animation/marionette/animation-controller.ts [hunk 1]
```diff
@@ -149,7 +150,24 @@ export class AnimationController extends Component {
...
+     * @param value @en Variable's value. @zh 变量的值。
+     * @example
+     * ```ts
+     * animationController.setValue('speed', 3.14);
+     * animationController.setValue('crouching', true);
+     * animationController.setValue('attack', true);
+     * ```
+     * @experimental
+     */
+    public setValue_experimental (name: string, value: Value) {
```
Comment (SantyWang) [inline cocos/animation/marionette/animation-controller.ts:170]: What's difference between `setValue` and `setValue_experimental`?

File: cocos/animation/marionette/variable/primitive-variable.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        this._type = type;
+        switch (type) {
+        default:
+            break;
+        case VariableType.FLOAT:
+            this._value = 0;
+            break;
+        case VariableType.INTEGER:
+            this._value = 0.0;
```
Comment (SantyWang) [inline cocos/animation/marionette/variable/primitive-variable.ts]: Why init it as 0.0 when its type is INTERGER

Comment (shrinktofit) [reply cocos/animation/marionette/variable/primitive-variable.ts]: Solved in https://github.com/cocos/cocos-engine/pull/14815/commits/51e3bbe33f83911d4adfa29e29ed45bc191168dd

File: cocos/animation/marionette/animation-controller.ts (hunk 1 above)
Comment (shrinktofit) [reply cocos/animation/marionette/animation-controller.ts:170]: If a user want to set vec3 variable, if he use `setValue`, he will receive a type error. Then he must use `setValue_experimental`, and he then knows the api is experimental.

Comment (shrinktofit) [reply cocos/animation/marionette/animation-controller.ts:170]: The idea is taken from https://github.com/jestjs/jest/blob/bc26cd79e60b7bf29d854293f0f011936fda1a5a/packages/jest-runtime/src/index.ts#L2138

**Review by SantyWang — APPROVED** [review]

**Pull request merged**